src/contracts/atlas/AtlasVerification.sol

```diff
@@ -58,129 +60,71 @@ contract AtlasVerification is EIP712, DAppIntegration {
-                    return (prunedSolverOps, ValidCallsResult.InvalidSequence);
+            // Check if past dapp's deadline
+            if (block.number > dAppOp.deadline) {
+                if (dAppOp.deadline != 0 && !isSimulation) {
+                    return (prunedSolverOps, ValidCallsResult.DAppDeadlineReached);
                 }
             }
-        } else if (dConfig.callConfig.allowsUnknownBundler()) {
-            // check if protocol allows unknown bundlers, and verify all signatures if they do
 
-            // check dapp signature
-            if (!_verifyDApp(dConfig, dAppOp, msgSender, isSimulation)) {
-                return (prunedSolverOps, ValidCallsResult.DAppSignatureInvalid);
+            // Some checks are only needed when call is not a simulation
+            if (isSimulation) {
+                // Add all solver ops if simulation
+                return (solverOps, ValidCallsResult.Valid);
             }
 
-            // check user signature
-            if (!_verifyUser(dConfig, userOp, msgSender, isSimulation)) {
-                return (prunedSolverOps, ValidCallsResult.UserSignatureInvalid);
+            // Check
+            if (dAppOp.bundler != address(0) && msgSender != dAppOp.bundler) {
+                return (prunedSolverOps, ValidCallsResult.InvalidBundler);
             }
 
-            // check callchainhash
-            if (dAppOp.callChainHash != CallVerification.getCallChainHash(dConfig, userOp, solverOps) && !isSimulation)
-            {
-                return (prunedSolverOps, ValidCallsResult.InvalidSequence);
+            // Check gas price is within user's limit
+            if (tx.gasprice > userOp.maxFeePerGas) {
+                return (prunedSolverOps, ValidCallsResult.GasPriceHigherThanMax);
             }
-        } else {
-            return (prunedSolverOps, ValidCallsResult.UnknownBundlerNotAllowed);
-        }
-
-        if (solverOps.length > type(uint8).max - 2) {
-            return (prunedSolverOps, ValidCallsResult.TooManySolverOps);
-        }
 
-        if (block.number > userOp.deadline) {
-            bool bypass = isSimulation && userOp.deadline == 0;
-            if (!bypass) {
-                return (prunedSolverOps, ValidCallsResult.UserDeadlineReached);
+            // Check that the value of the tx is greater than or equal to the value specified
+            if (msgValue < userOp.value) {
+                return (prunedSolverOps, ValidCallsResult.TxValueLowerThanCallValue);
             }
-        }
 
-        if (block.number > dAppOp.deadline) {
-            bool bypass = isSimulation && dAppOp.deadline == 0;
-            if (!bypass) {
-                return (prunedSolverOps, ValidCallsResult.DAppDeadlineReached);
+            // Check the Execution Environment address is a smart contract
+            if (executionEnvironment.codehash == bytes32(0)) {
+                return (prunedSolverOps, ValidCallsResult.ExecutionEnvEmpty);
             }
```

Comment:
Do we still need this check, considering that we're also going to be auto-deploying the EE contract at the start of a `metacall` tx if one has not yet been deployed?

Reply:
we don't, good catch

Reply:
updated and fixed